### PR TITLE
Reduce ABC complexity

### PR DESCRIPTION
**Pull request by buehmann**
I picked a few of the `Metrics/AbcSize` offenses from the todo list and reduced the complexity of the methods.

### CODE REVIEW

File: lib/rubocop/cop/style/copyright.rb [hunk 1]
```diff
@@ -57,6 +51,16 @@ def autocorrect_notice
...
         end
 
+        def verify_autocorrect_notice!
+          raise Warning, AUTOCORRECT_EMPTY_WARNING if autocorrect_notice.empty?
+
+          regex = Regexp.new(notice)
+          return if autocorrect_notice =~ regex
+
+          raise Warning, "AutocorrectNotice '#{autocorrect_notice}' must " \
+                          "match Notice /#{notice}/"
```
Comment (buehmann) [inline lib/rubocop/cop/style/copyright.rb]: A case for #6420 …

**Comment by bbatsov** [comment]
The changes look good to me.

**Comment by buehmann** [comment]
The failing specs seem to be due to a JRuby bug: https://github.com/jruby/jruby/issues/5935

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**